backend/src/controller.ts

```diff
@@ -284,4 +290,58 @@ export class Controller {
 
     return data
   }
+
+async getTransferTimes(params: any): Promise<TimeToBridgeStats> {
+  const { sourceChainSlug, destinationChainSlug } = params
+
+  if (!sourceChainSlug || !destinationChainSlug) {
+    return
+  }
+
+  const txTimes = await this.db.getTransferTimes(sourceChainSlug, destinationChainSlug)
+
+  // array of transfer times as numbers
+  const timesArray = txTimes.map(record => Number(record.bond_within_timestamp))
+
+  const cacheKey = `${sourceChainSlug}-${destinationChainSlug}`
+  const cacheDurationMs = 5 * 60 * 1000 // 5 minutes
+  const cachedStats = cache.get(cacheKey)
+
+  if (cachedStats && typeof cachedStats === 'object' && 'avg' in cachedStats && 'median' in cachedStats && 'percentile90' in cachedStats) {
+    return cachedStats as TimeToBridgeStats
+  }
+
+  const stats = timeToBridgeStats(timesArray)
+  cache.put(cacheKey, stats, cacheDurationMs)
+
+  return stats
+
```

Comment:
I think there's a closing brace missing here?

```suggestion
}
```